src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/stash/StashApiClient.java

```diff
@@ -436,6 +451,91 @@ public String call() throws Exception {
     return response;
   }
 
+  private String putRequest(String path, String status) throws UnsupportedEncodingException {
+    logger.log(Level.FINEST, "PR-PUT-REQUEST:" + path + " with: " + status);
+    HttpClient client = getHttpClient();
+    client.getState().setCredentials(AuthScope.ANY, credentials);
+
+    PutMethod httpput = new PutMethod(path);
+    httpput.setRequestHeader("Connection", "close");
+    httpput.setRequestHeader("X-Atlassian-Token", "no-check"); // xsrf
+
+    if (status != null) {
+      ObjectNode node = mapper.getNodeFactory().objectNode();
+      node.put("status", status);
+      StringRequestEntity requestEntity = null;
+      try {
+        requestEntity =
+            new StringRequestEntity(mapper.writeValueAsString(node), "application/json", "UTF-8");
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+      httpput.setRequestEntity(requestEntity);
+    }
+
+    String response = "";
+    FutureTask<String> httpTask = null;
+    Thread thread;
+
+    try {
+      httpTask =
+          new FutureTask<String>(
+              new Callable<String>() {
+
+                private HttpClient client;
+                private PutMethod httpput;
+
+                @Override
+                public String call() throws Exception {
+
+                  String response = "";
+                  int responseCode = HttpStatus.SC_INTERNAL_SERVER_ERROR;
+                  responseCode = client.executeMethod(httpput);
+                  if (!validResponseCode(responseCode)) {
+                    logger.log(
```

Comment:
do not log and throw at the same time
since all other errors are re-thrown I think this one should be too

Reply:
also the log and exception have different information -- why?